src/Development/IDE/LSP/Notifications.hs

```diff
@@ -69,4 +71,9 @@ setHandlersNotifications = PartialHandlers $ \WithMessage{..} x -> return x
             logInfo (ideLogger ide) $ "Files created or deleted: " <> msg
             modifyFileExists ide events
             setSomethingModified ide
+    ,LSP.didChangeWorkspaceFoldersNotificationHandler = withNotification (LSP.didChangeWorkspaceFoldersNotificationHandler x) $
+        \_ ide (DidChangeWorkspaceFoldersParams events) -> do
+            modifyWorkspaceFolders ide
+              $ HS.union      (foldMap (HS.singleton . parseWorkspaceFolder) (_added   events))
+              . HS.difference (foldMap (HS.singleton . parseWorkspaceFolder) (_removed events))
```

Comment:
Is the `difference` here the right way around? `difference a b` is `a \\ b` whereas I think you want `b \\ a`.

Reply:
Absolutely. Good catch, thanks!